.github/workflows/CI.yml

```diff
@@ -0,0 +1,59 @@
+name: Build and Test k-NN
+on:
+  push:
+    branches:
+      - master
+      - opendistro-*
+
+jobs:
+  Build-k-NN:
+    strategy:
+      matrix:
+        java: [12]
+
+    name: Build and Test k-NN Plugin
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout k-NN
+        uses: actions/checkout@v1
+        
+      - name: Setup Java ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      
+      - name: Run build
+        run: |
+          ./gradlew build
+          ls -ltr build/distributions/
+          
+      - name: Pull and Run Docker
+        run: |
+          plugin=`ls build/distributions/*.zip`
+          version=`echo $plugin|awk -F- '{print $3}'| cut -d. -f 1-3`
+          plugin_version=`echo $plugin|awk -F- '{print $3}'| cut -d. -f 1-4`
+          echo $version
+          cd ..
+          
+          if docker pull opendistroforelasticsearch/opendistroforelasticsearch:$version
+          then
+            echo "FROM opendistroforelasticsearch/opendistroforelasticsearch:$version" >> Dockerfile
+            echo "RUN if [ -d /usr/share/elasticsearch/plugins/opendistro_security ]; then /usr/share/elasticsearch/bin/elasticsearch-plugin remove opendistro_security; fi" >> Dockerfile
```

Comment:
Why do we need to remove security plugin?

Reply:
The ESRestTest Client uses http by default. Need to disable the security plugin to call the rest api over http.

Reply:
Lets add it as a comment.

Reply:
Added.